compiler/backend/src/org/jetbrains/kotlin/codegen/inline/InlineCodegen.kt

```diff
@@ -433,145 +426,64 @@ abstract class InlineCodegen<out T : BaseExpressionCodegen>(
         typeArguments: List<TypeParameterMarker>?,
         typeSystem: TypeSystemCommonBackendContext
     ): SMAPAndMethodNode {
-        val intrinsic = generateInlineIntrinsic(state, functionDescriptor, typeArguments, typeSystem)
+        val intrinsic = generateInlineIntrinsic(state, functionDescriptor, jvmSignature.asmMethod, typeArguments, typeSystem)
         if (intrinsic != null) {
-            return SMAPAndMethodNode(intrinsic, createDefaultFakeSMAP())
+            return SMAPAndMethodNode(intrinsic, SMAP(listOf()))
         }
-
-        var asmMethod = mapMethod(callDefault)
-        if (asmMethod.name.contains("-") &&
-            !state.configuration.getBoolean(JVMConfigurationKeys.USE_OLD_INLINE_CLASSES_MANGLING_SCHEME) &&
```

Comment:
@pyos This changes is not-trivial and looks like it's works a bit differently cause of gone USE_OLD_INLINE_CLASSES_MANGLING_SCHEME check. Could you add clarifications here.  //cc  @ilmirus please help with reviewing

NB: I can merge all commits before `JVM: remove some FunctionDescriptors from SourceCompilerForInline` (including)

Reply:
There are two things to note here:
1. in the IR backend, the inline class lowering applies the same logic to all (not just inline) functions at IR level, so here `classFileContainsMethod` should always return true and the branch should never be entered; I've already mentioned this above.
2. in the old backend, [ExpressionCodegen does the same to `jvmSignature` before constructing `PsiInlineCodegen`](https://github.com/JetBrains/kotlin/blob/master/compiler/backend/src/org/jetbrains/kotlin/codegen/ExpressionCodegen.java#L2958-L2969), so if `callDefault` is false this branch does nothing.

Reply:
Thank you for clarification